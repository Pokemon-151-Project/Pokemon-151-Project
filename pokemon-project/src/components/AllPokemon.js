import React from "react";
import SinglePokemon from "./SinglePokemon";

const AllPokemon = (props) => {
  //This is the data object containing all the pokemon's info
  const { pokemon } = props;
  return (
    <section className="all-pokemon">
      {pokemon.map((poke) => {
        // This returns 151 cards, each containing a different pokemon. So AllPokemon is the parent container for all these cards.
        return (
          <SinglePokemon
            poke={poke}
            key={poke.name}
            data-testid="single-pokemon"
          />
        );
      })}
    </section>
  );
};
export default AllPokemon;
