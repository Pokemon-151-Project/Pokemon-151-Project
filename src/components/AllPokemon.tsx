import React from "react";
import SinglePokemon from "./SinglePokemon";
import { Pokemon } from "../data";

type Props = {
	pokemonData: Pokemon[];
	isShiny: boolean;
};

const AllPokemon: React.FC<Props> = ({ pokemonData, isShiny }) => {
	return (
		<main className="all-pokemon">
			{pokemonData.map((poke) => (
				<SinglePokemon
					poke={poke}
					key={poke.id}
					data-testid="single-pokemon"
					isShiny={isShiny}
				/>
			))}
		</main>
	);
};

export default AllPokemon;
