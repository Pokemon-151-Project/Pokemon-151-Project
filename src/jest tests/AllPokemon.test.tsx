import AllPokemon from "../components/AllPokemon";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";
import { Pokemon } from "../data";
import React from "react";

//fake array with one pokemon for testing purposes
const fakePokemon: Pokemon[] = [
	{
		id: 1,
		num: "001",
		name: "Fake Pokemon Name",
		type: ["Ground"],
		height: "4m",
		weight: "50kg",
		dexEntry: "blah blah blah",
	},
];
//This is similar to the above array but with multiple pokemon for testing purposes
const multFakePokemon: Pokemon[] = [
	{
		id: 1,
		num: "001",
		name: "Fake Pokemon Name",
		type: ["Bug"],
		dexEntry: "blah blah blah",
		height: "4m",
		weight: "50kg",
	},
	{
		id: 2,
		num: "002",
		name: "Fake2",
		type: ["Fire"],
		dexEntry: "This pokemon does nothing",
		height: "5m",
		weight: "60kg",
	},
	{
		id: 3,
		num: "003",
		name: "Fake3",
		type: ["Ghost", "Ice"],
		dexEntry: "This pokemon is god",
		height: "6m",
		weight: "80kg",
	},
];

test("[1] AllPokemon renders without errors", () => {
	render(<AllPokemon pokemonData={fakePokemon} isShiny={false} />);
});

test("[2] AllPokemon renders correctly with certain props, then rerenders when those props change", () => {
	//renders first with empty props array, then renders with fakePokemon (defined above) to make sure it correctly renders with changed props.
	const { rerender } = render(<AllPokemon pokemonData={[]} isShiny={false} />);

	const oldFakeName = screen.queryByText(/fake pokemon name/i);
	expect(oldFakeName).not.toBeInTheDocument();

	rerender(<AllPokemon pokemonData={fakePokemon} isShiny={false} />);
	const newFakeName = screen.queryByText(/fake pokemon name/i);
	expect(newFakeName).toBeInTheDocument();
});

test("[3] Can render multiple pokemon", () => {
	//Passing in an array with multiple pokemon to make sure it renders all of them

	render(<AllPokemon pokemonData={multFakePokemon} isShiny={false} />);

	const name1 = screen.queryByText(/fake pokemon name/i);
	const name2 = screen.queryByText(/fake2/i);
	const name3 = screen.queryByText(/fake3/i);
	const name4 = screen.queryByText(/fake4/i);
	expect(name1).toBeVisible();
	expect(name2).toBeVisible();
	expect(name3).toBeVisible();
	expect(name4).toBeFalsy();
});

test("[4] Matches snapshot from 5.8.22", () => {
	const component = renderer.create(
		<AllPokemon pokemonData={multFakePokemon} isShiny={false} />
	);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
