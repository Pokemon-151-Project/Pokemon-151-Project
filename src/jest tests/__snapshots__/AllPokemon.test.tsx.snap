// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[4] Matches snapshot from 5.8.22 1`] = `
<main
  className="sc-fsYfxw fzpIUR"
>
  <article
    className="sc-beyTiQ eLajlk"
    data-testid="pokemon-card"
  >
    <h2
      className="sc-guDMob iTmCCz"
    >
      Fake Pokemon Name
       
      <span
        className="sc-dmyDmy hoHUsp"
      >
        #
        1
      </span>
    </h2>
    <h3
      className="sc-hLQTFJ gkRoEt"
    >
      Bug
    </h3>
    <h4
      className="sc-eDLKEg igxwic"
    >
      Height: 
      4m
       
      <br />
      Weight: 
      50kg
    </h4>
    <figure
      className="sc-jTQDJr gXLfqK"
    >
      <img
        alt="Fake Pokemon Name"
        className="sc-gLLuHO dfPMAj"
        data-testid="not-shiny"
        src="https://www.serebii.net/swordshield/pokemon/001.png"
      />
    </figure>
    <blockquote
      className="sc-iBdoyZ dTKkRP"
    >
      blah blah blah
    </blockquote>
  </article>
  <article
    className="sc-beyTiQ eLajlk"
    data-testid="pokemon-card"
  >
    <h2
      className="sc-guDMob iTmCCz"
    >
      Fake2
       
      <span
        className="sc-dmyDmy hoHUsp"
      >
        #
        2
      </span>
    </h2>
    <h3
      className="sc-hLQTFJ gkRoEt"
    >
      Fire
    </h3>
    <h4
      className="sc-eDLKEg igxwic"
    >
      Height: 
      5m
       
      <br />
      Weight: 
      60kg
    </h4>
    <figure
      className="sc-jTQDJr gXLfqK"
    >
      <img
        alt="Fake2"
        className="sc-gLLuHO dfPMAj"
        data-testid="not-shiny"
        src="https://www.serebii.net/swordshield/pokemon/002.png"
      />
    </figure>
    <blockquote
      className="sc-iBdoyZ dTKkRP"
    >
      This pokemon does nothing
    </blockquote>
  </article>
  <article
    className="sc-beyTiQ eLajlk"
    data-testid="pokemon-card"
  >
    <h2
      className="sc-guDMob iTmCCz"
    >
      Fake3
       
      <span
        className="sc-dmyDmy hoHUsp"
      >
        #
        3
      </span>
    </h2>
    <h3
      className="sc-hLQTFJ gkRoEt"
    >
      Ghost / Ice
    </h3>
    <h4
      className="sc-eDLKEg igxwic"
    >
      Height: 
      6m
       
      <br />
      Weight: 
      80kg
    </h4>
    <figure
      className="sc-jTQDJr gXLfqK"
    >
      <img
        alt="Fake3"
        className="sc-gLLuHO dfPMAj"
        data-testid="not-shiny"
        src="https://www.serebii.net/swordshield/pokemon/003.png"
      />
    </figure>
    <blockquote
      className="sc-iBdoyZ dTKkRP"
    >
      This pokemon is god
    </blockquote>
  </article>
</main>
`;
